### PR TITLE
[image-builder-api] log more, traces aren't always available

### PR DESCRIPTION
**Pull request by kylos101**
## Description

Some environments lack tracing. Do more logging to better understand image build flows in said environments.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at fa807d1</samp>

Improved logging and formatting of image builder API client. Added log messages to track image build requests and results in `sugar.ts`.

</details>

## Related Issue(s)

Fixes # n/a

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - kylos101-orvnwvgjfwo</li>
	<li><b>🔗 URL</b> - <a href="https://kylos101-orvnwvgjfwo.preview.gitpod-dev.com/workspaces" target="_blank">kylos101-orvnwvgjfwo.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - kylos101-ORVnWVgJFwo-gha.15866</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/workspace-starter.ts [hunk 1]
```diff
@@ -497,6 +497,12 @@ export class WorkspaceStarter {
     ): Promise<StartWorkspaceResult> {
         const span = TraceContext.startSpan("actuallyStartWorkspace", ctx);
         span.setTag("region_preference", region);
+        log.info("actuallyStartWorkspace interesting params", {
```
Comment (easyCZ) [inline components/server/src/workspace/workspace-starter.ts:500]:
```suggestion
        log.info("Attempting to start workspace", {
```

File: components/image-builder-api/typescript/src/sugar.ts [hunk 2]
```diff
@@ -235,6 +238,12 @@ export class PromisifiedImageBuilderClient {
                     }
                 }
 
+                log.info("Needs image build?", {
+                    status: resp.getStatus(),
+                    source: request.getSource(),
+                    forceRebuild: request.getForceRebuild(),
+                    triggeredBy: request.getTriggeredBy(),
```
Comment (easyCZ) [inline components/image-builder-api/typescript/src/sugar.ts:245]: What values does this field contain? Is it potentially sensitive?

**Review by easyCZ — APPROVED** [review]

**Comment by akosyakov** [comment]
Did you test that it actually logging what you need? Asking because of redaction. It would be nice to have a short background or reference in the PR description as well as test steps.

**Review by akosyakov — APPROVED** [review]

File: components/image-builder-api/typescript/src/sugar.ts (hunk 2 above)
Comment (kylos101) [reply components/image-builder-api/typescript/src/sugar.ts:245]: > Is it potentially sensitive?

It is not

**Comment by kylos101** [comment]
> Did you test that it actually logging what you need? 

:wave: @akosyakov indeed, you can see a couple examples [here](https://cloudlogging.app.goo.gl/fWiKLFVK6ceVcqXk8) and [here](https://cloudlogging.app.goo.gl/aQxNe8tBXxSqVjG79). Thank you for asking, I should have included for posterity.

> Asking because of redaction. 

There was a redacted field I missed and removed in [daa9ab0](https://github.com/gitpod-io/gitpod/pull/18579/commits/daa9ab0bcc95bb578a2aa9bc67bdaa291742f64c).

> It would be nice to have a short background or reference in the PR description as well as test steps.

For the future, when there are reports of start workspace trouble, I would like to have this info for image build attempts.

To test, I started an image build by incrementing [this](https://github.com/kylos101/gitpod-custom-image/blob/kylos101/image-build-work/.gitpod.Dockerfile#L3), and started workspaces in the preview environment.

**Pull request merged**